dev-python/loky/loky-2.9.0.ebuild

```diff
@@ -3,7 +3,7 @@
 
```

Comment:
Hmm, I'm getting tons of errors here :-(

```
____________________________ ERROR at setup of TestsProcessPoolForkserverShutdown.test_run_after_shutdown _____________________________

request = <FixtureRequest for <Function test_run_after_shutdown>>

    def fill(request):
        item = request._pyfuncitem
        fixturenames = getattr(item, "fixturenames", None)
        if fixturenames is None:
            fixturenames = request.fixturenames
    
        if hasattr(item, 'callspec'):
            for param, val in sorted_by_dependency(item.callspec.params, fixturenames):
                if val is not None and is_lazy_fixture(val):
                    item.callspec.params[param] = request.getfixturevalue(val.name)
                elif param not in item.funcargs:
                    item.funcargs[param] = request.getfixturevalue(param)
    
>       _fillfixtures()

_fillfixtures = <bound method FixtureRequest._fillfixtures of <FixtureRequest for <Function test_run_after_shutdown>>>
fixturenames = ['_fail_for_invalid_template_variable',
 'django_test_environment',
 '_session_faker',
 'django_db_blocker',
 '_django_setup_unittest',
 'xunit_setup_class_fixture_TestsProcessPoolForkserverShutdown',
 '_dj_autoclear_mailbox',
 '_django_clear_site_cache',
 '_django_db_marker',
 '_django_set_urlconf',
 '_live_server_helper',
 '_template_string_if_invalid_marker',
 'xunit_setup_method_fixture_TestsProcessPoolForkserverShutdown',
 'setup_method',
 'request']
item       = <Function test_run_after_shutdown>
request    = <FixtureRequest for <Function test_run_after_shutdown>>

/usr/lib/python3.8/site-packages/pytest_lazyfixture.py:39: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/_executor_mixin.py:150: in setup_method
    _check_subprocesses_number(self.executor, self.worker_count)
        self       = <tests.test_process_executor_forkserver.TestsProcessPoolForkserverShutdown object at 0x7fa6f090ae50>
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

executor = <loky.process_executor.ProcessPoolExecutor object at 0x7fa6f09aba90>, expected_process_number = 5
expected_max_process_number = None, patience = 100

    def _check_subprocesses_number(executor, expected_process_number=None,
                                   expected_max_process_number=None, patience=100):
        # Wait for terminating processes to disappear
        children_cmdlines = _running_children_with_cmdline(psutil.Process())
        pids_cmdlines = [(c.pid, cmdline) for c, cmdline in children_cmdlines]
        children_pids = set(pid for pid, _ in pids_cmdlines)
        if executor is not None:
            worker_pids = set(executor._processes.keys())
        else:
            # Bypass pids checks when executor has been garbage
            # collected
            worker_pids = children_pids
        if expected_process_number is not None:
            try:
>               assert len(children_pids) == expected_process_number, pids_cmdlines
E               AssertionError: [(80, 'Xvfb -br -nolisten tcp -screen 0 800x600x16 -displayfd 10'), (573, "/usr/bin/python3.8 -B -c from multiprocessi...usr/lib/python38.zip', '/usr/lib/python3.8', '/usr/lib/python3.8/lib-dynload', '/usr/lib/python3.8/site-packages']})")]
E               assert 6 == 5
E                 +6
E                 -5

children_cmdlines = [(psutil.Process(pid=80, name='Xvfb', status='sleeping', started='18:49:57'),
  'Xvfb -br -nolisten tcp -screen 0 800x600x16 -displayfd 10'),
 (psutil.Process(pid=573, name='python3.8', status='sleeping', started='18:50:32'),
  '/usr/bin/python3.8 -B -c from multiprocessing.forkserver import main; '
  "main(23, 24, ['__main__'], **{'sys_path': "
  "['/tmp/portage/dev-python/loky-2.9.0/work/loky-2.9.0', "
  "'/tmp/portage/dev-python/loky-2.9.0/work/loky-2.9.0-python3_8/lib', "
  "'/tmp/portage/dev-python/loky-2.9.0/work/loky-2.9.0', "
  "'/usr/lib/python38.zip', '/usr/lib/python3.8', "
  "'/usr/lib/python3.8/lib-dynload', '/usr/lib/python3.8/site-packages']})"),
 (psutil.Process(pid=574, name='python3.8', status='sleeping', started='18:50:32'),
  '/usr/bin/python3.8 -B -c from multiprocessing.forkserver import main; '
  "main(23, 24, ['__main__'], **{'sys_path': "
  "['/tmp/portage/dev-python/loky-2.9.0/work/loky-2.9.0', "
  "'/tmp/portage/dev-python/loky-2.9.0/work/loky-2.9.0-python3_8/lib', "
  "'/tmp/portage/dev-python/loky-2.9.0/work/loky-2.9.0', "
  "'/usr/lib/python38.zip', '/usr/lib/python3.8', "
  "'/usr/lib/python3.8/lib-dynload', '/usr/lib/python3.8/site-packages']})"),
 (psutil.Process(pid=575, name='python3.8', status='sleeping', started='18:50:32'),
  '/usr/bin/python3.8 -B -c from multiprocessing.forkserver import main; '
  "main(23, 24, ['__main__'], **{'sys_path': "
  "['/tmp/portage/dev-python/loky-2.9.0/work/loky-2.9.0', "
  "'/tmp/portage/dev-python/loky-2.9.0/work/loky-2.9.0-python3_8/lib', "
  "'/tmp/portage/dev-python/loky-2.9.0/work/loky-2.9.0', "
  "'/usr/lib/python38.zip', '/usr/lib/python3.8', "
  "'/usr/lib/python3.8/lib-dynload', '/usr/lib/python3.8/site-packages']})"),
 (psutil.Process(pid=576, name='python3.8', status='sleeping', started='18:50:32'),
  '/usr/bin/python3.8 -B -c from multiprocessing.forkserver import main; '
  "main(23, 24, ['__main__'], **{'sys_path': "
  "['/tmp/portage/dev-python/loky-2.9.0/work/loky-2.9.0', "
  "'/tmp/portage/dev-python/loky-2.9.0/work/loky-2.9.0-python3_8/lib', "
  "'/tmp/portage/dev-python/loky-2.9.0/work/loky-2.9.0', "
  "'/usr/lib/python38.zip', '/usr/lib/python3.8', "
  "'/usr/lib/python3.8/lib-dynload', '/usr/lib/python3.8/site-packages']})"),
 (psutil.Process(pid=577, name='python3.8', status='sleeping', started='18:50:32'),
  '/usr/bin/python3.8 -B -c from multiprocessing.forkserver import main; '
  "main(23, 24, ['__main__'], **{'sys_path': "
  "['/tmp/portage/dev-python/loky-2.9.0/work/loky-2.9.0', "
  "'/tmp/portage/dev-python/loky-2.9.0/work/loky-2.9.0-python3_8/lib', "
  "'/tmp/portage/dev-python/loky-2.9.0/work/loky-2.9.0', "
  "'/usr/lib/python38.zip', '/usr/lib/python3.8', "
  "'/usr/lib/python3.8/lib-dynload', '/usr/lib/python3.8/site-packages']})")]
children_pids = {576, 577, 80, 573, 574, 575}
executor   = <loky.process_executor.ProcessPoolExecutor object at 0x7fa6f09aba90>
expected_max_process_number = None
expected_process_number = 5
patience   = 100
pids_cmdlines = [(80, 'Xvfb -br -nolisten tcp -screen 0 800x600x16 -displayfd 10'),
 (573,
  '/usr/bin/python3.8 -B -c from multiprocessing.forkserver import main; '
  "main(23, 24, ['__main__'], **{'sys_path': "
  "['/tmp/portage/dev-python/loky-2.9.0/work/loky-2.9.0', "
  "'/tmp/portage/dev-python/loky-2.9.0/work/loky-2.9.0-python3_8/lib', "
  "'/tmp/portage/dev-python/loky-2.9.0/work/loky-2.9.0', "
  "'/usr/lib/python38.zip', '/usr/lib/python3.8', "
  "'/usr/lib/python3.8/lib-dynload', '/usr/lib/python3.8/site-packages']})"),
 (574,
  '/usr/bin/python3.8 -B -c from multiprocessing.forkserver import main; '
  "main(23, 24, ['__main__'], **{'sys_path': "
  "['/tmp/portage/dev-python/loky-2.9.0/work/loky-2.9.0', "
  "'/tmp/portage/dev-python/loky-2.9.0/work/loky-2.9.0-python3_8/lib', "
  "'/tmp/portage/dev-python/loky-2.9.0/work/loky-2.9.0', "
  "'/usr/lib/python38.zip', '/usr/lib/python3.8', "
  "'/usr/lib/python3.8/lib-dynload', '/usr/lib/python3.8/site-packages']})"),
 (575,
  '/usr/bin/python3.8 -B -c from multiprocessing.forkserver import main; '
  "main(23, 24, ['__main__'], **{'sys_path': "
  "['/tmp/portage/dev-python/loky-2.9.0/work/loky-2.9.0', "
  "'/tmp/portage/dev-python/loky-2.9.0/work/loky-2.9.0-python3_8/lib', "
  "'/tmp/portage/dev-python/loky-2.9.0/work/loky-2.9.0', "
  "'/usr/lib/python38.zip', '/usr/lib/python3.8', "
  "'/usr/lib/python3.8/lib-dynload', '/usr/lib/python3.8/site-packages']})"),
 (576,
  '/usr/bin/python3.8 -B -c from multiprocessing.forkserver import main; '
  "main(23, 24, ['__main__'], **{'sys_path': "
  "['/tmp/portage/dev-python/loky-2.9.0/work/loky-2.9.0', "
  "'/tmp/portage/dev-python/loky-2.9.0/work/loky-2.9.0-python3_8/lib', "
  "'/tmp/portage/dev-python/loky-2.9.0/work/loky-2.9.0', "
  "'/usr/lib/python38.zip', '/usr/lib/python3.8', "
  "'/usr/lib/python3.8/lib-dynload', '/usr/lib/python3.8/site-packages']})"),
 (577,
  '/usr/bin/python3.8 -B -c from multiprocessing.forkserver import main; '
  "main(23, 24, ['__main__'], **{'sys_path': "
  "['/tmp/portage/dev-python/loky-2.9.0/work/loky-2.9.0', "
  "'/tmp/portage/dev-python/loky-2.9.0/work/loky-2.9.0-python3_8/lib', "
  "'/tmp/portage/dev-python/loky-2.9.0/work/loky-2.9.0', "
  "'/usr/lib/python38.zip', '/usr/lib/python3.8', "
  "'/usr/lib/python3.8/lib-dynload', '/usr/lib/python3.8/site-packages']})")]
worker_pids = {576, 577, 573, 574, 575}

tests/_executor_mixin.py:84: AssertionError
-------------------------------------------------------- Captured stdout setup --------------------------------------------------------
setup class with <multiprocessing.context.ForkServerContext object at 0x7fa6fd209ac0>
-------------------------------------------------------- Captured stderr setup --------------------------------------------------------
[DEBUG:MainProcess:MainThread] created semlock with handle 140355214573568
[DEBUG:MainProcess:MainThread] created semlock with handle 140355212013568
[DEBUG:MainProcess:MainThread] created semlock with handle 140355211202560
[DEBUG:MainProcess:MainThread] created semlock with handle 140355211198464
[DEBUG:MainProcess:MainThread] created semlock with handle 140355210285056
[DEBUG:MainProcess:MainThread] created semlock with handle 140355210280960
[DEBUG:MainProcess:MainThread] created semlock with handle 140355209838592
[DEBUG:MainProcess:MainThread] created semlock with handle 140355209834496
[DEBUG:MainProcess:MainThread] created semlock with handle 140355209830400
[DEBUG:MainProcess:MainThread] Queue._after_fork()
[DEBUG:MainProcess:MainThread] created semlock with handle 140355209826304
[DEBUG:MainProcess:MainThread] created semlock with handle 140355205382144
[DEBUG:MainProcess:MainThread] ProcessPoolExecutor is setup
[DEBUG:MainProcess:MainThread] created semlock with handle 140355202629632
[DEBUG:MainProcess:MainThread] created semlock with handle 140355202625536
[DEBUG:MainProcess:MainThread] created semlock with handle 140355200393216
[DEBUG:MainProcess:MainThread] created semlock with handle 140355200389120
[DEBUG:MainProcess:MainThread] created semlock with handle 140355191242752
[DEBUG:MainProcess:MainThread] Adjust process count : {573: <ForkServerProcess name='ForkServerProcess-22' pid=573 parent=67 started>, 574: <ForkServerProcess name='ForkServerProcess-23' pid=574 parent=67 started>, 575: <ForkServerProcess name='ForkServerProcess-24' pid=575 parent=67 started>, 576: <ForkServerProcess name='ForkServerProcess-25' pid=576 parent=67 started>, 577: <ForkServerProcess name='ForkServerProcess-26' pid=577 parent=67 started>}
[DEBUG:MainProcess:MainThread] _start_executor_manager_thread called
[DEBUG:MainProcess:Thread-22] Queue._start_thread()
[DEBUG:MainProcess:Thread-22] doing self._thread.start()
[DEBUG:MainProcess:QueueFeederThread] starting thread to feed data to pipe
[DEBUG:MainProcess:Thread-22] ... done self._thread.start()
[DEBUG/ForkServerProcess-25] Using 'cloudpickle' for serialization.
[DEBUG/ForkServerProcess-24] Using 'cloudpickle' for serialization.
[DEBUG/ForkServerProcess-23] Using 'cloudpickle' for serialization.
[DEBUG/ForkServerProcess-25] recreated blocker with handle 140355209838592
[DEBUG/ForkServerProcess-25] recreated blocker with handle 140355209834496
[DEBUG/ForkServerProcess-25] recreated blocker with handle 140355209830400
[DEBUG/ForkServerProcess-25] Queue._after_fork()
[DEBUG/ForkServerProcess-25] recreated blocker with handle 140355209826304
[DEBUG/ForkServerProcess-25] recreated blocker with handle 140355205382144
[DEBUG/ForkServerProcess-22] Using 'cloudpickle' for serialization.
[DEBUG/ForkServerProcess-26] Using 'cloudpickle' for serialization.
[DEBUG/ForkServerProcess-23] recreated blocker with handle 140355209838592
[DEBUG/ForkServerProcess-23] recreated blocker with handle 140355209834496
[DEBUG/ForkServerProcess-23] recreated blocker with handle 140355209830400
[DEBUG/ForkServerProcess-23] Queue._after_fork()
[DEBUG/ForkServerProcess-23] recreated blocker with handle 140355209826304
[DEBUG/ForkServerProcess-23] recreated blocker with handle 140355205382144
[DEBUG/ForkServerProcess-24] recreated blocker with handle 140355209838592
[DEBUG/ForkServerProcess-24] recreated blocker with handle 140355209834496
[DEBUG/ForkServerProcess-24] recreated blocker with handle 140355209830400
[DEBUG/ForkServerProcess-24] Queue._after_fork()
[DEBUG/ForkServerProcess-24] recreated blocker with handle 140355209826304
[DEBUG/ForkServerProcess-24] recreated blocker with handle 140355205382144
[DEBUG/ForkServerProcess-26] recreated blocker with handle 140355209838592
[DEBUG/ForkServerProcess-26] recreated blocker with handle 140355209834496
[DEBUG/ForkServerProcess-26] recreated blocker with handle 140355209830400
[DEBUG/ForkServerProcess-26] Queue._after_fork()
[DEBUG/ForkServerProcess-26] recreated blocker with handle 140355209826304
[DEBUG/ForkServerProcess-26] recreated blocker with handle 140355205382144
[DEBUG/ForkServerProcess-22] recreated blocker with handle 140355209838592
[DEBUG/ForkServerProcess-22] recreated blocker with handle 140355209834496
[DEBUG/ForkServerProcess-22] recreated blocker with handle 140355209830400
[DEBUG/ForkServerProcess-22] Queue._after_fork()
[DEBUG/ForkServerProcess-22] recreated blocker with handle 140355209826304
[DEBUG/ForkServerProcess-22] recreated blocker with handle 140355205382144
[DEBUG/ForkServerProcess-25] recreated blocker with handle 140355214573568
[DEBUG/ForkServerProcess-25] recreated blocker with handle 140355212013568
[DEBUG/ForkServerProcess-25] recreated blocker with handle 140355211202560
[DEBUG/ForkServerProcess-25] recreated blocker with handle 140355211198464
[DEBUG/ForkServerProcess-25] recreated blocker with handle 140355210285056
[DEBUG/ForkServerProcess-25] recreated blocker with handle 140355210280960
[DEBUG/ForkServerProcess-25] recreated blocker with handle 140355200389120
[INFO/ForkServerProcess-25] child process calling self.run()
[DEBUG/ForkServerProcess-25] Worker started with timeout=None
```

Reply:
On my clean container it doesn't happen - so an interesting failure. I thought maybe some kind of a race condition - nothing after 10 tries and all passes.
Have found some bug report with some fixes upstream, so I tries to import i here. Does it help?

Also I noted that some tests are skipped on here, do they match the one at your side?
```
SKIPPED [1] tests/utils.py:184: Test requires parallel_sum to be compiled
SKIPPED [3] tests/_test_process_executor.py:908: --skip-high-memory option was provided
SKIPPED [3] tests/_test_process_executor.py:916: These Pythons can pickle objects of size > 2 ** 31GB
SKIPPED [2] tests/_test_process_executor.py:1026: Only work with loky context
```

Reply:
I'm going to try to figure it out. I'm thinking it could be some pytest plugin.

Reply:
Yep, it's that. Now just have to figure out which one ;-).